extensions/applicationinsights-debugplugin-js/src/InternPlugin.ts

```diff
@@ -0,0 +1,252 @@
+/**
+ * DebugPlugin.ts
+ * @copyright Microsoft 2020
+ */
+
+import {
+  BaseTelemetryPlugin, IConfiguration, CoreUtils,
+  IAppInsightsCore, IPlugin, ITelemetryItem, IProcessTelemetryContext, _InternalLogMessage, LoggingSeverity, _InternalMessageId, getNavigator,
+  ITelemetryPluginChain, InstrumentFunc, IInstrumentHooksCallbacks, IInstrumentCallDetails, InstrumentFuncs
+} from '@microsoft/applicationinsights-core-js';
+import { LoggerElement } from './components/helpers';
+import { tempStyle, permStyle } from './components/styleNodeSrc';
+import { DebugBin, DebugBinParent } from './components/debugBins';
+
+
+export default class DebugPlugin extends BaseTelemetryPlugin {
+  /**
+   * the root element of the logger
+   */
+  public myRootElement: HTMLDivElement;
+
+  /**
+   * the logging element
+   */
+  public myLogger: LoggerElement;
+
+  /**
+   * the style that is only on the webpage when the log is active
+   */
+  public myTempStyle: HTMLStyleElement;
+
+  /**
+   * the style that will be permanently embedded in the webpage
+   * TODO: manage style conflicts (prepend unique ID to relevant class names?)
+   */
+  public myPermStyle: HTMLStyleElement;
+
+  /**
+   * an object containing the individual debug bin items
+   */
+  public debugBins: {
+    [key: string]: DebugBin;
+  };
+
+  /**
+   * the different telemetry functions that will be tracked
+   */
+  public trackers?: string[];
+
+  /**
+   * appinsights analytics extension instance. useful methods on __proto__
+   */
+  public analyticsExt?: IPlugin;
+
+  /**
+   * ajax dependency extension instance. useful methods on __proto__
+   */
+  public ajaxDependencyExt?: IPlugin;
+
+  /**
+   * appinsights properties extension instance. useful methods on __proto__
+   */
+  public propertiesExt?: IPlugin;
+
+  /**
+   * appinsights channel extension instance.
+   */
+  public channelPluginExt?: IPlugin;
+
+  /**
+   * timestamp used to track number of seconds since webpage was loaded
+   */
+  public startTime = +new Date();
+
+  constructor(
+    trackers?: string[],
+  ) {
+    super();
+    this.trackers = trackers;
+  }
+
+  initialize(config: IConfiguration, core: IAppInsightsCore, extensions: IPlugin[], pluginChain?: ITelemetryPluginChain) {
```

Comment:
This method code is quite big, consider to split in multiple methods for better readability